In [1]:
import numpy as np
import pandas as pd
data_fl = pd.read_csv("FL_insurance_sample.csv")
data_fl

policyID statecode           county  eq_site_limit  hu_site_limit  \
0        119736        FL      CLAY COUNTY       498960.0      498960.00   
1        448094        FL      CLAY COUNTY      1322376.3     1322376.30   
2        206893        FL      CLAY COUNTY       190724.4      190724.40   
3        333743        FL      CLAY COUNTY            0.0       79520.76   
4        172534        FL      CLAY COUNTY            0.0      254281.50   
...         ...       ...              ...            ...            ...   
36629    404309        FL  PINELLAS COUNTY            0.0       88722.00   
36630    465875        FL  PINELLAS COUNTY      1297057.5     1297057.50   
36631    791209        FL  PINELLAS COUNTY       173286.9      173286.90   
36632    322627        FL  PINELLAS COUNTY      1499781.6     1499781.60   
36633    398149        FL  PINELLAS COUNTY       373488.3      373488.30   

       fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  \
0           498960.0       498960.0   498960.00   792148.90   
1          1322376.3      1322376.3  1322376.30  1438163.57   
2           190724.4       190724.4   190724.40   192476.78   
3                0.0            0.0    79520.76    86854.48   
4                0.0       254281.5   254281.50   246144.49   
...              ...            ...         ...         ...   
36629            0.0        88722.0    88722.00   155065.65   
36630      1297057.5      1297057.5  1297057.50  1577500.79   
36631            0.0            0.0   173286.90   205527.97   
36632            0.0            0.0  1499781.60  1768782.43   
36633            0.0            0.0   373488.30   596003.67   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
0                     0.0              9979.2                 0.0   
1                     0.0                 0.0                 0.0   
2                     0.0                 0.0                 0.0   
3                     0.0                 0.0                 0.0   
4                     0.0                 0.0                 0.0   
...                   ...                 ...                 ...   
36629                 0.0                 0.0                 0.0   
36630                 0.0                 0.0                 0.0   
36631                 0.0                 0.0                 0.0   
36632                 0.0                 0.0                 0.0   
36633                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
0                       0       30.102261       -81.711777  Residential   
1                       0       30.063936       -81.707664  Residential   
2                       0       30.089579       -81.700455  Residential   
3                       0       30.063236       -81.707703  Residential   
4                       0       30.060614       -81.702675  Residential   
...                   ...             ...              ...          ...   
36629                   0       28.122885       -82.770218  Residential   
36630                   0       28.080900       -82.758800  Residential   
36631                   0       28.110550       -82.766360  Residential   
36632                   0       28.089415       -82.697411  Residential   
36633                   0       28.064440       -82.774590  Residential   

      construction  point_granularity  
0          Masonry                  1  
1          Masonry                  3  
2             Wood                  1  
3             Wood                  3  
4             Wood                  1  
...            ...                ...  
36629         Wood                  3  
36630      Masonry                  1  
36631         Wood                  4  
36632      Masonry                  3  
36633      Masonry                  1  

[36634 rows x 18 columns]

In [3]:
#numpy version
print(np.__version__)
#pandas version
print(pd.__version__)

1.18.1
1.0.0


In [4]:
print(type(data_fl))

<class 'pandas.core.frame.DataFrame'>


In [5]:
print(data_fl.shape)

(36634, 18)


In [6]:
print(data_fl.shape[0])
print(data_fl.shape[1])

36634
18


In [8]:
print(data_fl.columns)
print(data_fl.index)

Index(['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity'],
      dtype='object')
RangeIndex(start=0, stop=36634, step=1)


In [9]:
print(data_fl.dtypes)
print(type(data_fl.dtypes))

policyID                int64
statecode              object
county                 object
eq_site_limit         float64
hu_site_limit         float64
fl_site_limit         float64
fr_site_limit         float64
tiv_2011              float64
tiv_2012              float64
eq_site_deductible    float64
hu_site_deductible    float64
fl_site_deductible    float64
fr_site_deductible      int64
point_latitude        float64
point_longitude       float64
line                   object
construction           object
point_granularity       int64
dtype: object
<class 'pandas.core.series.Series'>


In [10]:
print(data_fl.dtypes[0])

int64


In [11]:
print(data_fl.dtypes[-1])

int64


#### Observe data

In [14]:
data_fl.head(5)

policyID statecode       county  eq_site_limit  hu_site_limit  \
0    119736        FL  CLAY COUNTY       498960.0      498960.00   
1    448094        FL  CLAY COUNTY      1322376.3     1322376.30   
2    206893        FL  CLAY COUNTY       190724.4      190724.40   
3    333743        FL  CLAY COUNTY            0.0       79520.76   
4    172534        FL  CLAY COUNTY            0.0      254281.50   

   fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  eq_site_deductible  \
0       498960.0       498960.0   498960.00   792148.90                 0.0   
1      1322376.3      1322376.3  1322376.30  1438163.57                 0.0   
2       190724.4       190724.4   190724.40   192476.78                 0.0   
3            0.0            0.0    79520.76    86854.48                 0.0   
4            0.0       254281.5   254281.50   246144.49                 0.0   

   hu_site_deductible  fl_site_deductible  fr_site_deductible  point_latitude  \
0              9979.2                 0.0                   0       30.102261   
1                 0.0                 0.0                   0       30.063936   
2                 0.0                 0.0                   0       30.089579   
3                 0.0                 0.0                   0       30.063236   
4                 0.0                 0.0                   0       30.060614   

   point_longitude         line construction  point_granularity  
0       -81.711777  Residential      Masonry                  1  
1       -81.707664  Residential      Masonry                  3  
2       -81.700455  Residential         Wood                  1  
3       -81.707703  Residential         Wood                  3  
4       -81.702675  Residential         Wood                  1

In [16]:
data_fl.tail(4)

policyID statecode           county  eq_site_limit  hu_site_limit  \
36630    465875        FL  PINELLAS COUNTY      1297057.5      1297057.5   
36631    791209        FL  PINELLAS COUNTY       173286.9       173286.9   
36632    322627        FL  PINELLAS COUNTY      1499781.6      1499781.6   
36633    398149        FL  PINELLAS COUNTY       373488.3       373488.3   

       fl_site_limit  fr_site_limit   tiv_2011    tiv_2012  \
36630      1297057.5      1297057.5  1297057.5  1577500.79   
36631            0.0            0.0   173286.9   205527.97   
36632            0.0            0.0  1499781.6  1768782.43   
36633            0.0            0.0   373488.3   596003.67   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
36630                 0.0                 0.0                 0.0   
36631                 0.0                 0.0                 0.0   
36632                 0.0                 0.0                 0.0   
36633                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
36630                   0       28.080900       -82.758800  Residential   
36631                   0       28.110550       -82.766360  Residential   
36632                   0       28.089415       -82.697411  Residential   
36633                   0       28.064440       -82.774590  Residential   

      construction  point_granularity  
36630      Masonry                  1  
36631         Wood                  4  
36632      Masonry                  3  
36633      Masonry                  1

In [18]:
#by default it will print only 1 row
data_fl.sample()

policyID statecode          county  eq_site_limit  hu_site_limit  \
14705    215232        FL  BREVARD COUNTY            0.0            0.0   

       fl_site_limit  fr_site_limit   tiv_2011    tiv_2012  \
14705            0.0      2007180.0  2007180.0  2368271.68   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
14705                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude        line  \
14705                   0         28.3962         -80.6064  Commercial   

             construction  point_granularity  
14705  Reinforced Masonry                  4

In [19]:
# 4 rows randomly
data_fl.sample(4)

policyID statecode             county  eq_site_limit  hu_site_limit  \
32286    234228        FL   CHARLOTTE COUNTY            0.0        7663.24   
10572    902627        FL     ALACHUA COUNTY            0.0       67335.83   
23900    699124        FL  PALM BEACH COUNTY            0.0       14744.44   
20412    588445        FL  MIAMI DADE COUNTY            0.0     2962610.64   

       fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  \
32286            0.0            0.0     7663.24     9289.38   
10572            0.0            0.0    67335.83    92463.95   
23900            0.0            0.0    14744.44    14056.88   
20412            0.0            0.0  2962610.64  3491460.34   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
32286                 0.0                 0.0                 0.0   
10572                 0.0                 0.0                 0.0   
23900                 0.0                 0.0                 0.0   
20412                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
32286                   0       26.971525       -82.104060  Residential   
10572                   0       29.793991       -82.173271  Residential   
23900                   0       26.662340       -80.258186  Residential   
20412                   0       25.769820       -80.361680   Commercial   

             construction  point_granularity  
32286                Wood                  3  
10572                Wood                  1  
23900                Wood                  1  
20412  Reinforced Masonry                  1

In [22]:
type(data_fl.sample(4))

pandas.core.frame.DataFrame

#### Info about the dataframe

In [21]:
data_fl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36634 entries, 0 to 36633
Data columns (total 18 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   policyID            36634 non-null  int64  
 1   statecode           36634 non-null  object 
 2   county              36634 non-null  object 
 3   eq_site_limit       36634 non-null  float64
 4   hu_site_limit       36634 non-null  float64
 5   fl_site_limit       36634 non-null  float64
 6   fr_site_limit       36634 non-null  float64
 7   tiv_2011            36634 non-null  float64
 8   tiv_2012            36634 non-null  float64
 9   eq_site_deductible  36634 non-null  float64
 10  hu_site_deductible  36634 non-null  float64
 11  fl_site_deductible  36634 non-null  float64
 12  fr_site_deductible  36634 non-null  int64  
 13  point_latitude      36634 non-null  float64
 14  point_longitude     36634 non-null  float64
 15  line                36634 non-null  object 
 16  cons

In [26]:
data_fl.line.value_counts()

Residential    30838
Commercial      5796
Name: line, dtype: int64

In [27]:
data_fl.dtypes.value_counts()

float64    11
object      4
int64       3
dtype: int64

In [28]:
print(type(data_fl.info()))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36634 entries, 0 to 36633
Data columns (total 18 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   policyID            36634 non-null  int64  
 1   statecode           36634 non-null  object 
 2   county              36634 non-null  object 
 3   eq_site_limit       36634 non-null  float64
 4   hu_site_limit       36634 non-null  float64
 5   fl_site_limit       36634 non-null  float64
 6   fr_site_limit       36634 non-null  float64
 7   tiv_2011            36634 non-null  float64
 8   tiv_2012            36634 non-null  float64
 9   eq_site_deductible  36634 non-null  float64
 10  hu_site_deductible  36634 non-null  float64
 11  fl_site_deductible  36634 non-null  float64
 12  fr_site_deductible  36634 non-null  int64  
 13  point_latitude      36634 non-null  float64
 14  point_longitude     36634 non-null  float64
 15  line                36634 non-null  object 
 16  cons

#### Accessing a single column

In [30]:
type(data_fl.policyID)

pandas.core.series.Series

In [32]:
data_fl.policyID

0        119736
1        448094
2        206893
3        333743
4        172534
          ...  
36629    404309
36630    465875
36631    791209
36632    322627
36633    398149
Name: policyID, Length: 36634, dtype: int64

In [31]:
data_fl['policyID']

0        119736
1        448094
2        206893
3        333743
4        172534
          ...  
36629    404309
36630    465875
36631    791209
36632    322627
36633    398149
Name: policyID, Length: 36634, dtype: int64

#### specific column datatype

In [38]:
data_fl.policyID.dtypes

dtype('int64')

In [40]:
data_fl.dtype

AttributeError: 'DataFrame' object has no attribute 'dtype'

#### difference between dtypes and dtype
- 'DataFrame' object has no attribute 'dtype'

- Numerical columns-int or float data
    - mean,median,sum,quartiles,min,max
- Categorical columns - Object type or string data
- Boolean data- True or False

In [41]:
data_fl.columns

Index(['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity'],
      dtype='object')

In [42]:
data_fl.point_granularity.dtype

dtype('int64')

In [43]:
data_fl.point_granularity

0        1
1        3
2        1
3        3
4        1
        ..
36629    3
36630    1
36631    4
36632    3
36633    1
Name: point_granularity, Length: 36634, dtype: int64

In [44]:
# mean value
data_fl.point_granularity.mean()

1.6409073538243162

In [45]:
data_fl.point_granularity.shape[0]

36634

In [46]:
# average value
data_fl.point_granularity.sum()/data_fl.point_granularity.shape[0]

1.6409073538243162

In [48]:
#min value
print(data_fl.point_granularity.min())
#max value
print(data_fl.point_granularity.max())

1
7


In [49]:
data_fl.point_granularity.value_counts()

1    26569
3     6724
4     3337
7        2
5        2
Name: point_granularity, dtype: int64

In [50]:
#mode
data_fl.point_granularity.mode()

0    1
dtype: int64

In [52]:
data_fl.point_granularity.mode()[0]

1

In [53]:
data_fl.point_granularity.median()

1.0

In [54]:
#Range of value in a column
print(data_fl.point_granularity.min(),data_fl.point_granularity.max())

1 7


In [57]:
data_fl.county.value_counts()
#output type is a series
#series of labels are unique values
#the value for that label indicates number of times the label occured in that column
#value count is integer

MIAMI DADE COUNTY    4315
BROWARD COUNTY       3193
PALM BEACH COUNTY    2791
DUVAL COUNTY         1894
ORANGE COUNTY        1811
                     ... 
GLADES COUNTY          22
UNION COUNTY           15
Orlando                 1
OSCEOLA COUNTY          1
North Fort Myers        1
Name: county, Length: 67, dtype: int64

In [105]:
print(type(data_fl.county.count()))

<class 'numpy.int32'>


In [58]:
#to get how many non null values
data_fl.county.count()

36634

In [59]:
#to check number of unique values
data_fl.county.nunique()

67

In [60]:
type(data_fl.county.nunique())

int

In [61]:
type(data_fl.county.count())

numpy.int32

In [62]:
data_fl.county.value_counts().index

Index(['MIAMI DADE COUNTY', 'BROWARD COUNTY', 'PALM BEACH COUNTY',
       'DUVAL COUNTY', 'ORANGE COUNTY', 'PINELLAS COUNTY', 'POLK COUNTY',
       'VOLUSIA COUNTY', 'HILLSBOROUGH COUNTY', 'MARION COUNTY',
       'OKALOOSA COUNTY', 'SEMINOLE COUNTY', 'ALACHUA COUNTY',
       'BREVARD COUNTY', 'SANTA ROSA COUNTY', 'PASCO COUNTY', 'COLLIER COUNTY',
       'LEE COUNTY', 'ST  JOHNS COUNTY', 'MANATEE COUNTY', 'ESCAMBIA COUNTY',
       'SARASOTA COUNTY', 'CHARLOTTE COUNTY', 'BAY COUNTY', 'CITRUS COUNTY',
       'INDIAN RIVER COUNTY', 'HIGHLANDS COUNTY', 'CLAY COUNTY',
       'WALTON COUNTY', 'PUTNAM COUNTY', 'LEON COUNTY', 'JACKSON COUNTY',
       'LAKE COUNTY', 'FLAGLER COUNTY', 'GADSDEN COUNTY', 'SUMTER COUNTY',
       'SUWANNEE COUNTY', 'MONROE COUNTY', 'NASSAU COUNTY', 'LEVY COUNTY',
       'COLUMBIA COUNTY', 'HERNANDO COUNTY', 'WASHINGTON COUNTY',
       'TAYLOR COUNTY', 'MARTIN COUNTY', 'DESOTO COUNTY', 'WAKULLA COUNTY',
       'HARDEE COUNTY', 'MADISON COUNTY', 'HENDRY COUNTY', 'GULF 

In [63]:
data_fl.county.value_counts().shape[0]

67

In [65]:
data_fl.describe()

policyID  eq_site_limit  hu_site_limit  fl_site_limit  \
count   36634.000000   3.663400e+04   3.663400e+04   3.663400e+04   
mean   548661.937189   7.314780e+05   2.074348e+06   6.646009e+05   
std    259912.926494   1.752300e+07   1.964150e+07   1.746114e+07   
min    100074.000000   0.000000e+00   0.000000e+00   0.000000e+00   
25%    323647.750000   0.000000e+00   4.248106e+04   0.000000e+00   
50%    548525.000000   0.000000e+00   1.926911e+05   0.000000e+00   
75%    774252.000000   0.000000e+00   9.163381e+05   0.000000e+00   
max    999971.000000   2.160000e+09   2.160000e+09   2.160000e+09   

       fr_site_limit      tiv_2011      tiv_2012  eq_site_deductible  \
count   3.663400e+04  3.663400e+04  3.663400e+04        3.663400e+04   
mean    9.911724e+05  2.172875e+06  2.571004e+06        7.787908e+02   
std     1.767178e+07  2.015015e+07  2.036777e+07        4.787979e+04   
min     0.000000e+00  9.000000e+01  7.337000e+01        0.000000e+00   
25%     0.000000e+00  4.562415e+04  5.401351e+04        0.000000e+00   
50%     0.000000e+00  2.021051e+05  2.416310e+05        0.000000e+00   
75%     5.741010e+04  9.465821e+05  1.128871e+06        0.000000e+00   
max     2.160000e+09  2.160000e+09  1.701000e+09        6.273765e+06   

       hu_site_deductible  fl_site_deductible  fr_site_deductible  \
count        3.663400e+04        36634.000000        36634.000000   
mean         7.037985e+03          192.453210           26.483594   
std          9.582593e+04         8037.713716         4702.740932   
min          0.000000e+00            0.000000            0.000000   
25%          0.000000e+00            0.000000            0.000000   
50%          0.000000e+00            0.000000            0.000000   
75%          0.000000e+00            0.000000            0.000000   
max          7.380000e+06       450000.000000       900000.000000   

       point_latitude  point_longitude  point_granularity  
count    36634.000000     36634.000000       36634.000000  
mean        28.087477       -81.903565           1.640907  
std          1.647734         1.809066           1.070561  
min         24.547514       -87.447290           1.000000  
25%         26.471996       -82.439331           1.000000  
50%         28.057077       -81.585700           1.000000  
75%         29.623005       -80.372910           3.000000  
max         30.989820       -80.033257           7.000000

In [67]:
data_fl.describe().columns

Index(['policyID', 'eq_site_limit', 'hu_site_limit', 'fl_site_limit',
       'fr_site_limit', 'tiv_2011', 'tiv_2012', 'eq_site_deductible',
       'hu_site_deductible', 'fl_site_deductible', 'fr_site_deductible',
       'point_latitude', 'point_longitude', 'point_granularity'],
      dtype='object')

In [68]:
data_fl.describe(include=object)

statecode             county         line construction
count      36634              36634        36634        36634
unique         1                 67            2            5
top           FL  MIAMI DADE COUNTY  Residential         Wood
freq       36634               4315        30838        21581

In [70]:
data_fl.describe(include=np.int64)

policyID  fr_site_deductible  point_granularity
count   36634.000000        36634.000000       36634.000000
mean   548661.937189           26.483594           1.640907
std    259912.926494         4702.740932           1.070561
min    100074.000000            0.000000           1.000000
25%    323647.750000            0.000000           1.000000
50%    548525.000000            0.000000           1.000000
75%    774252.000000            0.000000           3.000000
max    999971.000000       900000.000000           7.000000

In [71]:
data_fl.describe(include=np.float64)

eq_site_limit  hu_site_limit  fl_site_limit  fr_site_limit  \
count   3.663400e+04   3.663400e+04   3.663400e+04   3.663400e+04   
mean    7.314780e+05   2.074348e+06   6.646009e+05   9.911724e+05   
std     1.752300e+07   1.964150e+07   1.746114e+07   1.767178e+07   
min     0.000000e+00   0.000000e+00   0.000000e+00   0.000000e+00   
25%     0.000000e+00   4.248106e+04   0.000000e+00   0.000000e+00   
50%     0.000000e+00   1.926911e+05   0.000000e+00   0.000000e+00   
75%     0.000000e+00   9.163381e+05   0.000000e+00   5.741010e+04   
max     2.160000e+09   2.160000e+09   2.160000e+09   2.160000e+09   

           tiv_2011      tiv_2012  eq_site_deductible  hu_site_deductible  \
count  3.663400e+04  3.663400e+04        3.663400e+04        3.663400e+04   
mean   2.172875e+06  2.571004e+06        7.787908e+02        7.037985e+03   
std    2.015015e+07  2.036777e+07        4.787979e+04        9.582593e+04   
min    9.000000e+01  7.337000e+01        0.000000e+00        0.000000e+00   
25%    4.562415e+04  5.401351e+04        0.000000e+00        0.000000e+00   
50%    2.021051e+05  2.416310e+05        0.000000e+00        0.000000e+00   
75%    9.465821e+05  1.128871e+06        0.000000e+00        0.000000e+00   
max    2.160000e+09  1.701000e+09        6.273765e+06        7.380000e+06   

       fl_site_deductible  point_latitude  point_longitude  
count        36634.000000    36634.000000     36634.000000  
mean           192.453210       28.087477       -81.903565  
std           8037.713716        1.647734         1.809066  
min              0.000000       24.547514       -87.447290  
25%              0.000000       26.471996       -82.439331  
50%              0.000000       28.057077       -81.585700  
75%              0.000000       29.623005       -80.372910  
max         450000.000000       30.989820       -80.033257

In [73]:
data_fl.describe(include='all')

policyID statecode             county  eq_site_limit  \
count    36634.000000     36634              36634   3.663400e+04   
unique            NaN         1                 67            NaN   
top               NaN        FL  MIAMI DADE COUNTY            NaN   
freq              NaN     36634               4315            NaN   
mean    548661.937189       NaN                NaN   7.314780e+05   
std     259912.926494       NaN                NaN   1.752300e+07   
min     100074.000000       NaN                NaN   0.000000e+00   
25%     323647.750000       NaN                NaN   0.000000e+00   
50%     548525.000000       NaN                NaN   0.000000e+00   
75%     774252.000000       NaN                NaN   0.000000e+00   
max     999971.000000       NaN                NaN   2.160000e+09   

        hu_site_limit  fl_site_limit  fr_site_limit      tiv_2011  \
count    3.663400e+04   3.663400e+04   3.663400e+04  3.663400e+04   
unique            NaN            NaN            NaN           NaN   
top               NaN            NaN            NaN           NaN   
freq              NaN            NaN            NaN           NaN   
mean     2.074348e+06   6.646009e+05   9.911724e+05  2.172875e+06   
std      1.964150e+07   1.746114e+07   1.767178e+07  2.015015e+07   
min      0.000000e+00   0.000000e+00   0.000000e+00  9.000000e+01   
25%      4.248106e+04   0.000000e+00   0.000000e+00  4.562415e+04   
50%      1.926911e+05   0.000000e+00   0.000000e+00  2.021051e+05   
75%      9.163381e+05   0.000000e+00   5.741010e+04  9.465821e+05   
max      2.160000e+09   2.160000e+09   2.160000e+09  2.160000e+09   

            tiv_2012  eq_site_deductible  hu_site_deductible  \
count   3.663400e+04        3.663400e+04        3.663400e+04   
unique           NaN                 NaN                 NaN   
top              NaN                 NaN                 NaN   
freq             NaN                 NaN                 NaN   
mean    2.571004e+06        7.787908e+02        7.037985e+03   
std     2.036777e+07        4.787979e+04        9.582593e+04   
min     7.337000e+01        0.000000e+00        0.000000e+00   
25%     5.401351e+04        0.000000e+00        0.000000e+00   
50%     2.416310e+05        0.000000e+00        0.000000e+00   
75%     1.128871e+06        0.000000e+00        0.000000e+00   
max     1.701000e+09        6.273765e+06        7.380000e+06   

        fl_site_deductible  fr_site_deductible  point_latitude  \
count         36634.000000        36634.000000    36634.000000   
unique                 NaN                 NaN             NaN   
top                    NaN                 NaN             NaN   
freq                   NaN                 NaN             NaN   
mean            192.453210           26.483594       28.087477   
std            8037.713716         4702.740932        1.647734   
min               0.000000            0.000000       24.547514   
25%               0.000000            0.000000       26.471996   
50%               0.000000            0.000000       28.057077   
75%               0.000000            0.000000       29.623005   
max          450000.000000       900000.000000       30.989820   

        point_longitude         line construction  point_granularity  
count      36634.000000        36634        36634       36634.000000  
unique              NaN            2            5                NaN  
top                 NaN  Residential         Wood                NaN  
freq                NaN        30838        21581                NaN  
mean         -81.903565          NaN          NaN           1.640907  
std            1.809066          NaN          NaN           1.070561  
min          -87.447290          NaN          NaN           1.000000  
25%          -82.439331          NaN          NaN           1.000000  
50%          -81.585700          NaN          NaN           1.000000  
75%          -80.372910          NaN          NaN           3.000000  
max         

- Slicing, subsetting, Boolean Indexing
    - postion based (iloc)
    - rows --> 0 to 36633
    - columns --> 0 to 17
    - label based indexing(loc)
- rows --> 0 to 36633
- columns --> ['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit', 'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012', 'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible', 'fr_site_deductible', 'point_latitude', 'point_longitude', 'line', 'construction', 'point_granularity']

In [75]:
data_fl.iloc[1,:] #row with row 1 and all the columns

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [77]:
data_fl.loc[1,:]

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [78]:
data_fl.iloc[1,1]

'FL'

In [80]:
data_fl.loc[1,"statecode"]

'FL'

In [81]:
data_fl.iloc[1,1:3] #stop index is exclusive

statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [85]:
data_fl.loc[1,'statecode':'county']

statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [104]:
print(data_fl.iloc[1:4,1:3])
print(type(data_fl.iloc[1:4,1:3]))

  statecode       county
1        FL  CLAY COUNTY
2        FL  CLAY COUNTY
3        FL  CLAY COUNTY
<class 'pandas.core.frame.DataFrame'>


In [103]:
print(data_fl.loc[1:4,'statecode':'county'])
print(type(data_fl.loc[1:4,'statecode':'county']))

  statecode       county
1        FL  CLAY COUNTY
2        FL  CLAY COUNTY
3        FL  CLAY COUNTY
4        FL  CLAY COUNTY
<class 'pandas.core.frame.DataFrame'>


In [89]:
data_fl.set_index('policyID')

statecode           county  eq_site_limit  hu_site_limit  \
policyID                                                            
119736          FL      CLAY COUNTY       498960.0      498960.00   
448094          FL      CLAY COUNTY      1322376.3     1322376.30   
206893          FL      CLAY COUNTY       190724.4      190724.40   
333743          FL      CLAY COUNTY            0.0       79520.76   
172534          FL      CLAY COUNTY            0.0      254281.50   
...            ...              ...            ...            ...   
404309          FL  PINELLAS COUNTY            0.0       88722.00   
465875          FL  PINELLAS COUNTY      1297057.5     1297057.50   
791209          FL  PINELLAS COUNTY       173286.9      173286.90   
322627          FL  PINELLAS COUNTY      1499781.6     1499781.60   
398149          FL  PINELLAS COUNTY       373488.3      373488.30   

          fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  \
policyID                                                         
119736         498960.0       498960.0   498960.00   792148.90   
448094        1322376.3      1322376.3  1322376.30  1438163.57   
206893         190724.4       190724.4   190724.40   192476.78   
333743              0.0            0.0    79520.76    86854.48   
172534              0.0       254281.5   254281.50   246144.49   
...                 ...            ...         ...         ...   
404309              0.0        88722.0    88722.00   155065.65   
465875        1297057.5      1297057.5  1297057.50  1577500.79   
791209              0.0            0.0   173286.90   205527.97   
322627              0.0            0.0  1499781.60  1768782.43   
398149              0.0            0.0   373488.30   596003.67   

          eq_site_deductible  hu_site_deductible  fl_site_deductible  \
policyID                                                               
119736                   0.0              9979.2                 0.0   
448094                   0.0                 0.0                 0.0   
206893                   0.0                 0.0                 0.0   
333743                   0.0                 0.0                 0.0   
172534                   0.0                 0.0                 0.0   
...                      ...                 ...                 ...   
404309                   0.0                 0.0                 0.0   
465875                   0.0                 0.0                 0.0   
791209                   0.0                 0.0                 0.0   
322627                   0.0                 0.0                 0.0   
398149                   0.0                 0.0                 0.0   

          fr_site_deductible  point_latitude  point_longitude         line  \
policyID                                                                     
119736                     0       30.102261       -81.711777  Residential   
448094                     0       30.063936       -81.707664  Residential   
206893                     0       30.089579       -81.700455  Residential   
333743                     0       30.063236       -81.707703  Residential   
172534                     0       30.060614       -81.702675  Residential   
...                      ...             ...              ...          ...   
404309                     0       28.122885       -82.770218  Residential   
465875                     0       28.080900       -82.758800  Residential   
791209                     0       28.110550       -82.766360  Residential   
322627                     0       28.089415       -82.697411  Residential   
398149                     0       28.064440       -82.774590  Residential   

         construction  point_granularity  
policyID                                  
119736        Masonry                  1  
448094        Masonry                  3  
206893           Wood                  1  
333743           Wood                  3  
172534           Wood                  1  

In [90]:
data_fl.head()

policyID statecode       county  eq_site_limit  hu_site_limit  \
0    119736        FL  CLAY COUNTY       498960.0      498960.00   
1    448094        FL  CLAY COUNTY      1322376.3     1322376.30   
2    206893        FL  CLAY COUNTY       190724.4      190724.40   
3    333743        FL  CLAY COUNTY            0.0       79520.76   
4    172534        FL  CLAY COUNTY            0.0      254281.50   

   fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  eq_site_deductible  \
0       498960.0       498960.0   498960.00   792148.90                 0.0   
1      1322376.3      1322376.3  1322376.30  1438163.57                 0.0   
2       190724.4       190724.4   190724.40   192476.78                 0.0   
3            0.0            0.0    79520.76    86854.48                 0.0   
4            0.0       254281.5   254281.50   246144.49                 0.0   

   hu_site_deductible  fl_site_deductible  fr_site_deductible  point_latitude  \
0              9979.2                 0.0                   0       30.102261   
1                 0.0                 0.0                   0       30.063936   
2                 0.0                 0.0                   0       30.089579   
3                 0.0                 0.0                   0       30.063236   
4                 0.0                 0.0                   0       30.060614   

   point_longitude         line construction  point_granularity  
0       -81.711777  Residential      Masonry                  1  
1       -81.707664  Residential      Masonry                  3  
2       -81.700455  Residential         Wood                  1  
3       -81.707703  Residential         Wood                  3  
4       -81.702675  Residential         Wood                  1

In [108]:
#row with index 1 and all the columns
data_fl.iloc[1,:]

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [112]:
data_fl.loc[1,:]

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [113]:
# row with index 1 and column with index 1
data_fl.iloc[1,1]

'FL'

In [116]:
data_fl.loc[1,'point_granularity']

3

In [117]:
# row with index 1 and columns with index 1,2
data_fl.iloc[1,1:3]

statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [120]:
data_fl.loc[1,'policyID':'county']

policyID          448094
statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [126]:
# rows with indices 1,2,3
# columns with indices 1,2
data_fl.iloc[1:4,1:3]

statecode       county
1        FL  CLAY COUNTY
2        FL  CLAY COUNTY
3        FL  CLAY COUNTY

In [124]:
data_fl.loc[1:4,'policyID':'statecode']

policyID statecode
1    448094        FL
2    206893        FL
3    333743        FL
4    172534        FL

In [101]:
# rows with indices 1,2,3
# columns with indices 1,2
df4 = pd.DataFrame(data_fl, columns= ['policyID','point_granularity'], index=[1,2,3])
print(df4)
print(type(df4))

   policyID  point_granularity
1    448094                  3
2    206893                  1
3    333743                  3
<class 'pandas.core.frame.DataFrame'>
